app/controllers/admin/users_controller.rb

```diff
@@ -30,16 +30,17 @@ def update
 
     # PUT /users/1/become
     def become
-      sign_in(@user)
-      redirect_to request.referrer, notice: "Logged in as #{@user.username}"
+      sign_in @user
+      redirect_to root_url, notice: "Logged in as #{@user.name}"
```

Comment:
Hm you are right... I'll make it redirect to referrer if set or root if unset.

Reply:
Ah ok I remembered... this action is an admin action where you become (log in as) another user... So we can't use the referrer, since that would send you to the admin console and potentially raise an error due to lack of permissions for the new user. Sending the (admin) user to root is the easiest thing that should always work.

Reply:
sounds good.